File: OOAD_HW1/Q4/animal.java
package Q4;

/* Animal Super-class
 * Eating, sleeping, waking, and roaming do not seem to be very different across animals
 * in a zoo, so those four actions are handled here. The animal families will handle 
 * makeNoise(), since animals do make different sounds.
 */

public abstract class animal {
	
	// Since methods are going to be called at different steps of the inheritance, we
	// need to have a name all the way up here.
	protected String name;
	
	public animal() {
		this.name = "None";
	}
	
	public animal(String name) {
		this.name = name;
	}
	
	// Only requirement is a text output, so all that is needed is the animal's name
	// and what kind of animal it is to confirm that the animal is doing something.
	public void roam() {
		//animalName = aName;
		System.out.println(this.name + " is exercising.");
	}
	
	public void eat() {
		//animalName = aName;
		System.out.println(this.name + " is eating.");
	}
	
	public void sleep() {
		//animalName = aName;
		System.out.println(this.name + " goes to sleep, Zzz.");
	}
	
	public void wakeUp() {
		//animalName = aName;
		System.out.println(this.name + " has woken up.");
	}
	
	// makeNoise is overwritten everywhere else, but the compiler demands that
	// I have a method here.
	public void makeNoise() {
		System.out.println(this.name + " makes a generic noise.");
	}
}

File: OOAD_HW2/Part C/Animal.cpp
// Animal Superclass

#include <iostream>
#include <string>
#include <cstdlib>
#include <iomanip>

using namespace std;

/* Animal Super-class
 * Eating, sleeping, waking, and roaming do not seem to be very different across animals
 * in a zoo, so those four actions are handled here. The animal families will handle
 * makeNoise(), since animals do make different sounds.
 */

class Animal
{
    protected:
        string name;
        string type;

    public:
        void roam(void);
        void eat(void);
        void sleep(void);
        void wakeUp(void);
        void makeNoise(void);

        Animal(string aName, string aType);
};

Animal::Animal(string aName, string aType)
{
    name = aName;
    type = aType;   // Didn't need to do this, but was too late to rewrite everything with this member.
}


// Text outputs
void Animal::roam(void)
{
    cout << name << ", the " << type << " is exercising." << endl;
}

void Animal::eat(void)
{
    cout << name << ", the " << type << " is eating." << endl;
}

void Animal::sleep(void)
{
   cout << name << ", the " << type << " is sleeping." << endl;
}

void Animal::wakeUp(void)
{
    cout << name << ", the " << type << " is waking up." << endl;
}

void Animal::makeNoise(void)
{
    cout << name << ", the " << type << " makes a generic noise." << endl;
}
